crates/storage-query-datafusion/src/partition/schema.rs

```diff
@@ -8,30 +8,52 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
-#![allow(dead_code)]
-
 use datafusion::arrow::datatypes::DataType;
 
 use crate::table_macro::*;
 
 define_table!(
-    /// Partition placements and their target running mode
+    /// Partition table
     partition(
         /// Partition ID
         partition_id: DataType::UInt32,
 
-        /// Plain node ID where the partition is running
-        plain_node_id: DataType::Utf8,
-
-        /// Target run mode of partition (LEADER, FOLLOWER)
-        target_mode: DataType::Utf8,
-
         /// Partition start key
         start_key: DataType::UInt64,
 
         /// Partition end key
         end_key: DataType::UInt64,
 
+        /// The log id serving this partition
+        log_id: DataType::UInt32,
+
+        /// The column family name
+        cf_name: DataType::Utf8,
+
+        /// The database name
+        db_name: DataType::Utf8,
+
+        /// The partition leader generational node id acquired from gossip
+        leader_gen_node_id: DataType::Utf8,
+
+        /// The partition leader plain node id acquired from gossip
+        leader_plain_node_id: DataType::Utf8,
+
+        /// The observed leader epoch from gossip
+        leader_epoch: DataType::Utf8,
+
+        /// The observed current membership version from gossip
+        v_current: DataType::UInt32,
+
+        /// A comma separated list of nodes representing the current members of the replica-set
+        current_replica_set: DataType::Utf8,
+
+        /// The observed next membership version from gossip
+        v_next: DataType::UInt32,
+
+        /// A comma separated list of nodes representing the next members of the replica-set
+        next_replica_set: DataType::Utf8,
+
         /// Current known metadata version
         metadata_ver: DataType::UInt32,
```

Comment:
Should we rename this into partition table version? I was for a second confused which metadata version this field refers to. On the other hand, it is in the partition table where it makes sense that this is the partition table version.